.github/workflows/sonarcloud.yml

```diff
@@ -8,7 +8,22 @@ on:
     types: [opened, synchronize, reopened]
     path-ignore: ['abstractions/**', 'core/**', '**.md', '.vscode/**', '**.svg']
 jobs:
+  checksecret:
+    name: check if SONAR_TOKEN is set in github secrets
+    runs-on: ubuntu-latest
+    outputs:
+      is_SONAR_TOKEN_set: ${{ steps.checksecret_job.outputs.is_SONAR_TOKEN_set }}
+    steps:
+      - name: Check whether unity activation requests should be done
+        id: checksecret_job
+        env:
+            SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
Consider defining `SONAR_TOKEN` as a global env variable across the entire workflow since it is used in more than one job i.e. `checksecret` and `build`.

Reply:
thanks for the suggestion. updated